riptable/rt_fastarray.py

```diff
@@ -3382,7 +3382,8 @@ def to_arrow(self, type: Optional['pa.DataType'] = None, *, preserve_fixed_bytes
                 # Convert this array to a pyarrow variable-length string array.
                 if type is None:
                     type = pa.string()
-                return pa.array(self._np, mask=invalids_mask, type=type)
+                # Convert to binaryarray then to stringarray to preserve length (riptable#249)
+                return pa.array(self._np, mask=invalids_mask).cast(type)

```

Comment:
This may well work around the issue, but what I don't understand is how it'd be able to correctly handle all ASCII / Unicode cases. Is pyarrow just decoding the binary array elements as UTF-8? If so, some sequences of valid ASCII characters won't be valid UTF-8 and I'm guessing they would cause the pyarrow decoder to raise an exception.
If that's the case, you can preserve correctness (at the cost of some performance) by writing a check -- perhaps in numba -- that you'd apply only to ASCII string arrays and which would scan the strings to look for these incompatible sequences; if any are found, you could do something like ``self._np.astype('U')`` instead, then passing that to ``pa.array()`` instead of just ``self._np``.

Reply:
Note that this code path is when the input is bytes, which I presume to mean ASCII. So it really does reinterpret the buffer to be an array of int8's (with proper length, no less) which is then re-cast to be a string type.

Reply:
Per our conversation I thought of a safer alternative: converting the byte string into a Unicode string via numpy, then uing the (working) Unicode to pyarrow conversion.